### DISCUSSION TITLE
`brew edit` still shows a warning when EDITOR environmental variable is set

### DISCUSSION THREAD

**Discussion by whitebear60** [Homebrew/discussions#2886] [Tap maintenance and Homebrew development] [answered]
When I'm trying to edit formulae/casks code with `brew edit <formula>`, Homebrew shows me the warning
```
Warning: Using code because no editor was set in the environment.
This may change in the future, so we recommend setting EDITOR,
or HOMEBREW_EDITOR to your preferred text editor.
```
I've set those variables:
```
$ echo $EDITOR
/Applications/Visual Studio Code.app/Contents/Resources/app/bin/code

$ echo $HOMEBREW_EDITOR
/Applications/Visual Studio Code.app/Contents/Resources/app/bin/code
```
The warning still persists. Maybe I've done something wrong?

**Comment by carlocab** [comment, marked as answer]
Hmm, that's strange. Can you diff the outputs of
```
brew config
EDITOR= HOMEBREW_EDITOR= brew config
EDITOR= brew config
HOMEBREW_EDITOR= brew config
```
?

If you're not sure how to do that, you can just dump the output of each command here.

**Reply by whitebear60** [reply]
`brew config` doesn't seems to read HOMEBREW_EDITOR
```
HOMEBREW_VERSION: 3.3.12-53-g2ae21a5
ORIGIN: https://github.com/Homebrew/brew
HEAD: 2ae21a582a90571abb66b7fbceb04cd1e53785bf
Last commit: 2 days ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 31e286ff8652ba7daed1eda2069cf14644c22d21
Core tap last commit: 19 hours ago
Core tap branch: master
HOMEBREW_PREFIX: /usr/local
HOMEBREW_CASK_OPTS: []
HOMEBREW_CORE_GIT_REMOTE: https://github.com/Homebrew/homebrew-core
HOMEBREW_GITHUB_API_TOKEN: set
HOMEBREW_MAKE_JOBS: 4
Homebrew Ruby: 2.6.8 => /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.8/bin/ruby
CPU: quad-core 64-bit kabylake
Clang: 11.0.0 build 1100
Git: 2.21.1 => /Applications/Xcode.app/Contents/Developer/usr/bin/git
Curl: 7.54.0 => /usr/bin/curl
macOS: 10.14.6-x86_64
CLT: 11.3.1.0.1.1576735732
Xcode: 11.3.1
```

**Reply by whitebear60** [reply]
Outputs of `HOMEBREW_EDITOR= brew config`, `EDITOR= brew config`, `EDITOR= HOMEBREW_EDITOR= brew config` are identical and the same as I provided above

**Reply by whitebear60** [reply]
macOS 10.14.6, zsh 5.3, if that matters

**Reply by carlocab** [reply]
It should. I wonder if it's just confused by the space in the path. Try adding `/Applications/Visual Studio Code.app/Contents/Resources/app/bin` to your `PATH` and then set `EDITOR` or `HOMEBREW_EDITOR` to `code`.

**Reply by carlocab** [reply]
Though, if you installed VS Code with Homebrew, you should already have a `/usr/local/bin/code` so it should be enough to set `EDITOR` and/or `HOMEBREW_EDITOR` to `code`.

**Reply by whitebear60** [reply]
No, I installed VSCode by downloading it manually from official website...
Installed VSCode with Homebrew and it works fine... Thanks anyway for your help
